sdk/python/lib/pulumi/automation/_stack.py

```diff
@@ -78,9 +78,15 @@ def __init__(
         self.config: ConfigMap = {}
         for key in config:
             config_value = config[key]
-            self.config[key] = ConfigValue(
-                value=config_value["value"], secret=config_value["secret"]
+            secret = config_value["secret"]
+            # If it is a secret, and we're not showing secrets, the value is excluded from the JSON results.
+            # In that case, we'll just use the sentinal `[secret]` value. Otherwise, we expect to get a value.
+            value = (
+                config_value.get("value", "[secret]")
+                if secret
+                else config_value["value"]
             )
+            self.config[key] = ConfigValue(value=value, secret=secret)
```

Comment:
Pre-existing bug I ran into this when running the test against a stack with a secret config value. The problem is that `--json` does not include a `value` property in the emitted JSON when not showing secrets, which was causing the previous implementation to raise an error because the `value` key was not found.

The fix: if it's a secret, try to get the value (show_secrets=True), otherwise (show_secrets=False), fallback to `[secret]`.